Models/Classifier2D.py

```diff
@@ -5,21 +5,32 @@
 from monai.networks import nets
 from Models.UnetEncoder import UnetEncoder
 
+
 class Classifier2D(pl.LightningModule):
-    def __init__(self, config):
+    def __init__(self, config, module_str):
 
         super().__init__()
-        self.config   = config
-        self.backbone = getattr(models, config['MODEL']['Backbone'])
-        if 'densenet' in config['MODEL']['Backbone']:
-            self.backbone = self.backbone(pretrained=config['MODEL']['Pretrained'],drop_rate=config['MODEL']['Drop_Rate'])
-        else: self.backbone = self.backbone(pretrained=config['MODEL']['Pretrained'])            
-
-    def forward(self, x):
-        return self.backbone(x)
+        self.config = config
+        # if 'densenet' in config['MODEL']['Backbone']:
+        #     self.backbone = self.backbone(pretrained=config['MODEL']['Pretrained'],
+        #                                   drop_rate=config['MODEL']['Drop_Rate'])
+        # else:
+        #     self.backbone = self.backbone(pretrained=config['MODEL']['Pretrained'])
 
-    def weights_init(self, m):
-        if isinstance(m, nn.Conv2d) or isinstance(m, nn.Linear):
-            nn.init.xavier_uniform_(m.weight.data)
+        model = config['MODEL'][module_str + '_Backbone']
+        parameters = config[module_str + '_MODEL_PARAMETERS']
```

Comment:
I've personally changed your model to the one that you remved. Please don't change it back. The reason I'm using it is that it is simple and can load a majority of what we need. We don't need UNet encoder as it  is a synonym for any ConVnet or VIT.